testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/deploymentoverlay/jar/OverlayUtils.java

```
@@ -0,0 +1,139 @@
+        addr.add(ModelDescriptionConstants.DEPLOYMENT_OVERLAY, overlayName);
+        addr.add(ModelDescriptionConstants.DEPLOYMENT, deployment);
+        op.get(ModelDescriptionConstants.OP_ADDR).set(addr);
+        op.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.ADD);
+        ManagementOperations.executeOperation(managementClient.getControllerClient(), op);
+
+        op = new ModelNode();
+        addr = new ModelNode();
+        addr.add(ModelDescriptionConstants.DEPLOYMENT, deployment);
+        op.get(ModelDescriptionConstants.OP_ADDR).set(addr);
+        op.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.REDEPLOY);
+        ManagementOperations.executeOperation(managementClient.getControllerClient(), op);
+    }
+
+    public static void setupOverlay(final ManagementClient managementClient, final String deployment,final String overlayName, final String overlayPath,final String overlayedContent) throws Exception {
+        setupOverlay(managementClient, deployment, overlayName, Collections.singletonMap(overlayPath, overlayedContent));
+    }
+
+    public static void removeOverlay(final ManagementClient managementClient, final String deployment,final String overlayName, final Set<String> overlayPaths) throws Exception {
+        //TODO: proper cleanup
+        try {
+            for (String overlayPath: overlayPaths) {
+                removeContentItem(managementClient, overlayName, overlayPath);
+            }
+            removeDeploymentItem(managementClient, overlayName, deployment);
+
+            ModelNode op = new ModelNode();
+            op.get(ModelDescriptionConstants.OP_ADDR).set(ModelDescriptionConstants.DEPLOYMENT_OVERLAY, overlayName);
+            op.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.REMOVE);
+            ManagementOperations.executeOperation(managementClient.getControllerClient(), op);
+        } catch (Exception e) {
+            e.printStackTrace();
```

exception is swallowed.